flow/project.py

```diff
@@ -1699,6 +1714,71 @@ def _store_bundled(self, operations):
                     file.write(operation.id + '\n')
             return bid
 
+    def _fn_aggregate(self, id):
+        "Return the canonical name to store aggregate information."
+        return os.path.join(self.root_directory(), '.aggregate', id)
+
+    def _store_aggregates(self, operations):
+        """Store aggregate information on a per operation basis.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
+        for operation in operations:
+            op_fn = '{}.txt'.format(operation.name)
+            aggregate_wid = '{} {}'.format(operation.id, ' '.join(map(str, operation._jobs)))
+            fn_aggregate = self._fn_aggregate(op_fn)
+            os.makedirs(os.path.dirname(fn_aggregate), exist_ok=True)
+            if os.path.exists(fn_aggregate):
+                with open(fn_aggregate, 'r+') as file:
+                    agg_file_contents = file.read()
+                    if aggregate_wid in agg_file_contents.splitlines():
+                        continue
+                    else:
+                        file.write(aggregate_wid + '\n')
+            else:
+                with open(fn_aggregate, 'w') as file:
+                    file.write(aggregate_wid + '\n')
+
+    def _fetch_aggregates(self, group):
+        """Fetch submitted aggregates for a group.
+
+        This enables the fetching status of lost aggregates.
+
+        :param group:
+            FlowGroup associated with the operation.
+        :type group:
+            list :py:class:`flow.FlowGroup`
+        :yields:
+            All aggregates which were stored during submission.
+        """
+        dir = '.aggregates/{}.txt'.format(group.name)
+        if os.path.exists(dir):
+            with open(dir, 'r') as file:
+                for obj in file:
+                    aggregate = []
+                    _ids = obj.split(' ')
+                    submission_id = _ids[0]
+                    job_ids = _ids[1:-1]
+                    try:
+                        for job_id in job_ids:
+                            aggregate.append(self.open_job(id=job_id))
+
+                        aggregate = tuple(aggregate)
+                        # Checking whether the aggregate and the submission id match.
+                        # If not, then a user must have changed the submission id.
+                        # Hence skip this aggregate.
+                        if group._generate_id(aggregate) == submission_id:
+                            yield aggregate
```

Comment:
I'm also starting to think that if users change then this should be on them but for now, I'm removing this whole process of storing aggregates in this PR and this will be implemented in the next PR.